From bb256b999a1278f988084acb92684d3c049ae7bf Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:04:47 -0400
Subject: [powerpc] Make device tree accesses in HVC VIO console endian safe

Message-id: <1410545655-205645-59-git-send-email-dzickus@redhat.com>
Patchwork-id: 94628
O-Subject: [RHEL7 PATCH 058/626] powerpc: Make device tree accesses in HVC VIO console endian safe
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 1502b480e2296359728e4b4e112faf5a5599ca32
Author: Anton Blanchard <anton@samba.org>
Date:   Wed Aug 7 02:01:39 2013 +1000

    powerpc: Make device tree accesses in HVC VIO console endian safe

    Signed-off-by: Anton Blanchard <anton@samba.org>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/tty/hvc/hvc_vio.c b/drivers/tty/hvc/hvc_vio.c
index 0c62980..c791b18 100644
--- a/drivers/tty/hvc/hvc_vio.c
+++ b/drivers/tty/hvc/hvc_vio.c
@@ -404,7 +404,7 @@ module_exit(hvc_vio_exit);
 void __init hvc_vio_init_early(void)
 {
  struct device_node *stdout_node;
- const u32 *termno;
+ const __be32 *termno;
  const char *name;
  const struct hv_ops *ops;
 
@@ -429,7 +429,7 @@ void __init hvc_vio_init_early(void)
  termno = of_get_property(stdout_node, "reg", NULL);
  if (termno == NULL)
   goto out;
- hvterm_priv0.termno = *termno;
+ hvterm_priv0.termno = of_read_number(termno, 1);
  spin_lock_init(&hvterm_priv0.buf_lock);
  hvterm_privs[0] = &hvterm_priv0;
 
-- 
1.7.1